plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeColumnHandle.java

```diff
@@ -55,46 +57,49 @@
     public static final String FILE_MODIFIED_TIME_COLUMN_NAME = "$file_modified_time";
     public static final Type FILE_MODIFIED_TIME_TYPE = TIMESTAMP_TZ_MILLIS;
 
-    private final String name;
-    private final Type type;
+    private final String baseColumnName;
+    private final Type baseType;
     private final OptionalInt fieldId;
     // Hold field names in Parquet files
     // The value is same as 'name' when the column mapping mode is none
     // The value is same as 'delta.columnMapping.physicalName' when the column mapping mode is id or name. e.g. col-6707cc9e-f3aa-4e6b-b8ef-1b03d3475680
-    private final String physicalName;
+    private final String basePhysicalColumnName;
     // Hold type in Parquet files
     // The value is same as 'type' when the column mapping mode is none
     // The value is same as 'delta.columnMapping.physicalName' when the column mapping mode is id or name. e.g. row(col-5924c8b3-04cf-4146-abb5-2c229e7ff708 integer)
-    private final Type physicalType;
+    private final Type basePhysicalType;
     private final DeltaLakeColumnType columnType;
+    private final Optional<DeltaLakeColumnProjectionInfo> projectionInfo;
 
     @JsonCreator
     public DeltaLakeColumnHandle(
-            @JsonProperty("name") String name,
-            @JsonProperty("type") Type type,
+            @JsonProperty("baseColumnName") String baseColumnName,
+            @JsonProperty("baseType") Type baseType,
             @JsonProperty("fieldId") OptionalInt fieldId,
-            @JsonProperty("physicalName") String physicalName,
-            @JsonProperty("physicalType") Type physicalType,
-            @JsonProperty("columnType") DeltaLakeColumnType columnType)
+            @JsonProperty("basePhysicalColumnName") String basePhysicalColumnName,
+            @JsonProperty("basePhysicalType") Type basePhysicalType,
+            @JsonProperty("columnType") DeltaLakeColumnType columnType,
```

Comment:
is it base column type, or leaf column type, or both?

Reply:
It's leaf column type.

For table `testdelta`
```
 CREATE TABLE testdelta (
         a integer,
         r1 ROW(r2 ROW(a integer, b integer), c integer),
         d integer
      ) with (partitioned_by=ARRAY['a']);
```

`a` -> `PARTITION`
`r1.r2.b` -> `REGULAR`
`r1.c` -> `REGULAR`
`$path` -> `SYNTHESIZED`
`d` -> `REGULAR`
`r1` -> `REGULAR`

Currently `PARTITION` and `SYNTHESIZED` will always be one-level fields.